internal/core/data/event.go

```diff
@@ -139,6 +139,10 @@ func addNewEvent(
 			return "", err
 		}
 		e.ID = id
+		savedEvent, err := dbClient.EventById(id)
```

Comment:
Would it be better to simply set `created` prior to calling AddEvent so we don't have to make a call to read back the Event from the DB?

Reply:
Alternatively, why don't we have AddEvent return the event it inserted? That's more standard isn't it?

Reply:
@lenny-intel, updated the fix based on your suggestion to set the `created` timestamp before calling the `dbClient.AddEvent` function.

@rsdmike , the suggestion sounds great and we'll implement in the v2 API changes. Currently the change is included in https://github.com/edgexfoundry/edgex-go/pull/2610.